### PR TITLE
Faster history points load

### LINKED ISSUE #108: More efficient sim.output.history_points.load
It would be nice to save the result of `sim.output.history_points.load` and read only the new lines in the .his file.

This would dramatically improve the performance of `sim.output.history_points.load` (https://github.com/exabl/snek5000/blob/main/src/snek5000/output/history_points.py#L59).

### PR DESCRIPTION
**Pull request by paugier**
Fixes #108

### CODE REVIEW

**Comment by paugier** [comment]
We would really need a tests using cbox. I don't understand why it did not work last time we tried.

@ashwinvis Do you understand ?

Why do we need that https://github.com/exabl/snek5000/blob/main/tests/conftest.py#L171 ?

**Comment by ashwinvis** [comment]
@paugier In the CI I set `export SNEK_DEBUG=1` and then run `pytest`. This would also add GFortran flags which perform extra floating point checks like if NaN is encountered anywhere.

**Comment by ashwinvis** [comment]
IIRC, testing snek5000-cbox produced some error due to those flags. You might need to look at why that happens.

Also, if we are going to add snek5000-cbox as a test dependency, I would like to have a tagged version since it is in rapid development.

File: src/snek5000/output/history_points.py [hunk 1]
```diff
@@ -57,12 +59,66 @@ def _complete_params_with_default(cls, params):
...
+                line_coord = file.readline()
+                for _ in range(nb_points - 1):
+                    file.readline()
+                line_data = file.readline()
+
+            nb_chars_read = (
+                len(first_line)
+                + nb_points * len(line_coord)
+                + nb_data_lines_read * len(line_data)
+            )
```
Comment (ashwinvis) [inline src/snek5000/output/history_points.py:82]: Can we keep track of `nb_chars_read` in a class attribute and then seek to the position at once?

Comment (paugier) [reply src/snek5000/output/history_points.py:82]: I don't think it's necessary. It would be less clear and in some corner cases less robust.

**Comment by paugier** [comment]
Arg, locally, the cbox tests pass with `export SNEK_DEBUG=1`!

**Comment by paugier** [comment]
@ashwinvis Do you understand the error? I don't find the cause of the error in the log.

Locally, I can't reproduce.

**Comment by ashwinvis** [comment]
@paugier I will take a look.

**Comment by ashwinvis** [comment]
@paugier Does this make sense:

https://github.com/exabl/snek5000/runs/4252368682?check_suite_focus=true#step:8:5622

File: src/snek5000/assets/compiler.smk [hunk 2]
```diff
@@ -46,7 +46,7 @@ rule compile:
         exe="nek5000",
     shell:
         """
-        {params.make} -j {output.exe} | tee -a build.log
+        {params.make} -j1 {output.exe} | tee -a build.log
```
Comment (ashwinvis) [inline src/snek5000/assets/compiler.smk:49]: Revert bec884252035e0c152e1f0649bace306a072549f before merging

**Comment by ashwinvis** [comment]
It could be because it is not packaged correctly... take a look at:

https://github.com/exabl/snek5000-phill/blob/d249ab84f9147f7a255f204dee40ea03a663024f/setup.cfg#L61-L62

**Comment by paugier** [comment]
It should be fine now. Thanks for the help!

I reverted the `-j1` to `-j`.

`gfortran_log` is not tested but it could be done in a later PR!

**Comment by ashwinvis** [comment]
@paugier Thanks to you too! This and #101 would make a really powerful feature. The `gfortran_log` needs some more care -- better integration and function names -- so yes, it can wait.

I will do a little bit of polishing of this PR and merge it soon. It appears that the tag `stable` has not been pushed to GitHub. Can you do that?

**Review by ashwinvis — COMMENTED** [review]
Mark the tag `stable` in `requirements-tests.txt` when done

**Comment by ashwinvis** [comment]
I still don't see any tags https://github.com/exabl/snek5000-cbox/tags

**Comment by paugier** [comment]
I used a stable branch: https://github.com/exabl/snek5000-cbox/tree/stable
